espnet/nets/pytorch_backend/e2e_asr_transformer.py

```diff
@@ -181,14 +190,24 @@ def forward(self, xs_pad, ilens, ys_pad):
 
         # TODO(karita) show predected text
         # TODO(karita) calculate these stats
-        cer, cer_ctc, wer = 0.0, 0.0, 0.0
-        if self.ctc is None:
+        if self.mtlalpha == 0.0:
             loss_ctc = None
+            cer_ctc = None
         else:
             batch_size = xs_pad.size(0)
             hs_len = hs_mask.view(batch_size, -1).sum(1)
             loss_ctc = self.ctc(hs_pad.view(batch_size, -1, self.adim), hs_len, ys_pad)
 
+            ys_hat = self.ctc.argmax(hs_pad.view(batch_size, -1, self.adim)).data
+            cer_ctc = self.error_calculator(ys_hat.cpu(), ys_pad.cpu(), is_ctc=True)
+
+        # 5. compute cer/wer
+        if self.training or self.error_calculator is None:
+            cer, wer = None, None
+        else:
+            ys_hat = pred_pad.argmax(dim=-1)
+            cer, wer = self.error_calculator(ys_hat.cpu(), ys_pad.cpu())
```

Comment:
I think these `cer, wer` can be obtained even if `args.mtlalpha == 0`. If so, remove `if args.mtlapha > 0` in `__init__`.

Reply:
The `if` in main allows to avoid adding the `error_calculator` if there is no `report_cer` in `args` and the `mtlalpha` is equal to zero. I will change the way to add `error_calculator` for `False` cases and `mtlalpha=0` then.

Reply:
Why can `args` have no `eport_cer`? It is statically added in `asr_train.py`.

Reply:
Yes, but when I wrote the code for transformer I migrated it from the `e2e_asr` where is almost the same `if report_cer is not in vars(args)`:
https://github.com/espnet/espnet/blob/fcf4c8ceeb413cea333ed98dc6c9fb5339bf1150/espnet/nets/pytorch_backend/e2e_asr.py#L121-L134

I will modify this directly in #774

Reply:
Please do not spread this... You can do the right thing here because it does not have a breaking change. I think you can also delete `self.recog_args` in Transformer.

Reply:
got it.